internal/yggdrasil/yggdrasil.go

```diff
@@ -195,7 +200,10 @@ func (h *Handler) PostDataMessageForDevice(ctx context.Context, params yggdrasil
 			if errors.IsNotFound(err) {
 				contentJson, _ := json.Marshal(msg.Content)
 				registrationInfo := models.RegistrationInfo{}
-				json.Unmarshal(contentJson, &registrationInfo)
+				err := json.Unmarshal(contentJson, &registrationInfo)
+				if err != nil {
+					return nil
```

Comment:
That also should be HTTP 400 instead of `nil`

Reply:
indeed. for sure not nil. I added 400 to swagger.